File: 0283-move-zeroes/0283-move-zeroes.py
class Solution(object):
    def moveZeroes(self, nums):
        i = 0
        count = 0
        while i < len(nums):
            if nums[i] == 0:
                nums.pop(i)
                count += 1
            else:
                i += 1

        while count:
            nums.append(0)
            count -= 1
        
        return nums

File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int n = nums.size();
        vector<int> res(n, 0);

        int top = 0;
        for (int i = 0; i < n; i++) {
            if (nums[i] != 0) res[top++] = nums[i];
        }

        nums = res;
    }
};